### DISCUSSION TITLE
limit stide of second step of stepping up and down on stair

### DISCUSSION THREAD

**Discussion by Naoki-Hiraoka** [YutaKojio/hrpsys-base#43] [Q&A] [answered]
https://github.com/YutaKojio/hrpsys-base/blob/a6f1a9f1217d43f91fe477b013d79058b85ff5d3/rtc/AutoBalancer/GaitGenerator.cpp#L817-L827

この処理は、階段を登るときに次の段にぶつかったり、降りるときに踏み外したりするのを防ぐためのものですか? (道路の車道と歩道の段差のような、1段しかない段差を乗り越える場合には不要ですか? また、視覚がある場合には不要ですか?)

**Comment by YutaKojio** [comment, marked as answer]
実機実験が失敗することがあって，とりあえずすぐ上手くいくようにいろいろ小手先の修正をしていたものの一部になります．

目的は２つあって，
1. 段差がもともと不安定だったので，踏み出し幅を小さくして少しでも外乱の影響を小さくしようとした．
2. 段差を登るときに段差に引っ掛けることがあった．

で，
１に関してはvision関連のバグを直してきているのでもういらないかもしれない．
2に関しては遊脚軌道を途中の干渉回避も考えてway_pointを設定するようにすればよくて，こっちはTODOです．

> 道路の車道と歩道の段差のような、1段しかない段差を乗り越える場合には不要ですか?

１段でも関係あります．

> 視覚がある場合には不要ですか?

視覚ありのときだけ影響する部分です．
